### DISCUSSION TITLE
Cloning with git issues

### DISCUSSION THREAD

**Discussion by rdfry** [community/community#133585] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am having the username, password issue when using git to clone a public repository. Why is it asking me for those when it has been deprecated and its a public repository? I read about setting up ssh but it still makes no sense to me.

**Comment by davevad93** [comment]
Hi @rdfry ,

In general, password authentication is required when performing `git` operations regardless of whether it's a public or private repo.

When performing git operations with `HTTPS` it prompts you to enter your GitHub credentials every time you pull or push a repo and the password has to be a `Personal Access Token` (with the `repo` scopes selected) instead of your account password because as you rightly pointed out, support for password authentication is deprecated, for more info see the docs:

- [Why is Git always asking for my password?](https://docs.github.com/en/get-started/getting-started-with-git/why-is-git-always-asking-for-my-password)
-  [Managing your personal access tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)

On the other hand, when performing git operations with `SSH` such as `git clone` or `git push` you will be prompted to enter just the password you set when generating the `ssh key`, see the documentation for more information:

- [Generating a new SSH key and adding it to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

**Comment by KuyaMecky** [comment]
It seems Git is asking for a username and password when cloning a public repository, which is unusual. Since GitHub has deprecated password authentication, you should use a personal access token (PAT) if you're using HTTPS. Generate a PAT from GitHub and use it instead of your password. If you're using SSH, ensure your SSH key is properly set up by running `ssh -T git@github.com`. Check your repository URL to ensure it's correct (e.g., `https://github.com/username/repository.git` for HTTPS or `git@github.com:username/repository.git` for SSH). For further troubleshooting, verify your Git configuration and clear any cached credentials.
